model/kubernetes/kubernetes_cluster.rb

```diff
@@ -60,6 +71,93 @@ def self.kubeconfig(vm)
   def kubeconfig
     self.class.kubeconfig(cp_vms.first)
   end
+
+  def add_finalizer(svc)
+    kubectl("-n #{svc["metadata"]["namespace"]} patch service #{svc["metadata"]["name"]} --type='merge' -p '{\"metadata\":{\"finalizers\":[\"#{KubernetesCluster::SERVICE_FINALIZER}\"]}}'")
+  end
+
+  def remove_finalizer(svc)
+    kubectl("-n #{svc["metadata"]["namespace"]} patch service #{svc["metadata"]["name"]} --type='json' -p='[{\"op\": \"remove\", \"path\": \"/metadata/finalizers\",\"value\":[\"#{KubernetesCluster::SERVICE_FINALIZER}\"]}]'")
+  end
+
+  def set_load_balancer_host_name(svc, hostname)
+    kubectl("-n #{svc["metadata"]["namespace"]} patch service #{svc["metadata"]["name"]} --type='merge' -p '{\"status\":{\"loadBalancer\":{\"ingress\":[{\"hostname\":\"#{hostname}\"}]}}}' --subresource='status'")
+  end
+
+  def lb_name_for_svc
+    ubid
+  end
+
+  # Returning an array in this format:
+  # [[src_port_0, dst_port_0], [src_port_1, dst_port_1],...]
+  def lb_desired_ports(svc)
+    svc["spec"]["ports"].map { |port| [port["port"], port["nodePort"]] }
+  end
+
+  def is_deleted(svc)
+    svc.dig("metadata", "deletionTimestamp")
+  end
+
+  def extra_and_missing_vms_for_lb(lb)
```

Comment:
nit: rename to something like `vm_diff_for_lb` (using `diff` instead of `extra and missing`)? This can also apply to `_ports_`.